src/Framework/BuildCheck/BuildCheckEventArgs.cs

```diff
@@ -30,6 +30,17 @@ internal BuildCheckTracingEventArgs()
     {
     }
 
+    internal BuildCheckTracingEventArgs(Dictionary<string, TimeSpan> data, bool isAggregatedGlobalReport) : this(data)
+    {
+        IsAggregatedGlobalReport = isAggregatedGlobalReport;
+    }
+
+    /// <summary>
+    /// When true, the tracing information is from the whole build for logging purposes
+    /// When false, the tracing is being used for communication between nodes and central process
+    /// </summary>
+    public bool IsAggregatedGlobalReport { get; private set; } = false;
+
     public Dictionary<string, TimeSpan> TracingData { get; private set; } = tracingData;
```

Comment:
Does this need to be serialized to the binlog?

Reply:
It does not, but we use this event args to send stats during the build for communication between worker nodes and main node, and at the end of the build to get all stats. If we do not have this bool we continue adding stats after the end of the build and it crashes.

Reply:
I believe this was refering to the whole class. Yes we want it (https://github.com/dotnet/msbuild/pull/10016#discussion_r1568748145). It'll need to be handled in https://github.com/dotnet/msbuild/blob/main/src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs and https://github.com/dotnet/msbuild/blob/main/src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs in order to have those in a binlog (otherwise we'd have just the unstructured textual outputs there - those would then need to be string parsed with code tightly dependent on actual output format)